### PR TITLE
feat: Set degradation preference on video senders

### PR DESCRIPTION
**Pull request by jallamsetty1**
- Use 'maintain-framerate' for camera tracks and 'maintain-resolution' for desktop tracks.
- Apply cached video constraints/preferences on video unmute. Since the sender encodings are available only after the renegotiation is complete (on chrome/safari), constraints/preferences have to be applied after renegotiation is complete

### CODE REVIEW

File: modules/proxyconnection/ProxyConnectionPC.js [hunk 1]
```diff
@@ -232,31 +232,13 @@ export default class ProxyConnectionPC {
...
-        };
-
         /**
          * Create an instance of {@code RTC} as it is required for peer
          * connection creation by {@code JingleSessionPC}. An existing instance
          * of {@code RTC} from elsewhere should not be re-used because it is
          * a stateful grouping of utilities.
          */
-        this._rtc = new RTC(conferenceStub, {});
+        this._rtc = new RTC(this, {});
```
Comment (paweldomas) [inline modules/proxyconnection/ProxyConnectionPC.js:241]: why this change?

Comment (paweldomas) [reply modules/proxyconnection/ProxyConnectionPC.js:241]: https://github.com/jitsi/lib-jitsi-meet/commit/4e9319b10acc9a3abf58d51bd7512174673b5a39

Comment (jallamsetty1) [reply modules/proxyconnection/ProxyConnectionPC.js:241]: I am reverting the change you had to make here when I first added the handlers for TRACK_ADDED and TRACK_MUTE_CHANGED events. https://github.com/jitsi/lib-jitsi-meet/blob/master/modules/RTC/TraceablePeerConnection.js#L359
This is not needed anymore.

Comment (paweldomas) [reply modules/proxyconnection/ProxyConnectionPC.js:241]: But even without that code, what is `this` referring to? RTC takes conference, so lets keep the `conferenceStub` there, but make it empty and without the FIXME comment.

Comment (jallamsetty1) [reply modules/proxyconnection/ProxyConnectionPC.js]: oh I misread the comment in the code there. Added the empty stub.
